### PR TITLE
[MM-49692][MM-49183] Switch webContents URL checks over to the mm-desktop URLs

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
When I changed the URL scheme I introduced a few bugs when checking for the current `webContents`' URL:
- Resizing the window would cause the back bar to show no matter what
- We would always be reloading the other views when the first one loads.
- Logging in multiple tabs wouldn't work

This PR switches all those calls over to `mm-desktop:` and fixes up some tests that were disabled as well.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49692
https://mattermost.atlassian.net/browse/MM-49183

```release-note
NONE
```

### CODE REVIEW

**Review by m1lt0n — APPROVED** [review]

File: src/main/app/initialize.ts [hunk 1]
```diff
@@ -424,7 +424,7 @@ function initializeAfterAppReady() {
         const requestingURL = webContents.getURL();
 
         // is the requesting url trusted?
-        callback(urlUtils.isTrustedURL(requestingURL, Config.teams));
+        callback(urlUtils.isTrustedURL(requestingURL, Config.teams, urlUtils.parseURL(requestingURL)?.protocol === 'mm-desktop:'));
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/main/app/initialize.ts:427]: Yes I think that's a good idea. I'll add that here.

File: src/main/utils.ts [hunk 2]
```diff
@@ -205,3 +205,7 @@ export function createCookieSetDetailsFromCookieString(cookie: string, url: stri
         expirationDate: new Date(cookieObject.Expires).valueOf(),
     };
 }
+
+export function convertURLToMMDesktop(url: URL) {
+    return new URL(`${url}`.replace(/^http(s)?:/, 'mm-desktop:'));
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/main/utils.ts:210]: I tried doing this before and it straight up didn't work. I can try it again but I think for some reason you can't assign it in Electron.

[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/main/utils.ts]: The URL is creating just fine, but it's not letting me set the protocol.

[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/main/utils.ts]: I'm gonna merge as-is we can look into this as a follow-up

**Comment by devinbinnie** [comment]
> Should we request a review from Security on this?

I think we're good on this, they're planning on doing a bigger review anyways.

**Pull request merged**